frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.tsx

```diff
@@ -14,23 +14,45 @@ import { ButtonVariant } from '@patternfly/react-core'
 import { ArrowCircleUpIcon, ExternalLinkAltIcon } from '@patternfly/react-icons'
 import { Fragment, ReactNode, useState } from 'react'
 import { useTranslation } from '../../../../../lib/acm-i18next'
-import { useRecoilState } from 'recoil'
+import { useRecoilState, useRecoilValue,waitForAll } from 'recoil'
 import { ansibleJobState } from '../../../../../atoms'
 import { RbacButton } from '../../../../../components/Rbac'
 import { rbacCreate, rbacPatch } from '../../../../../lib/rbac-util'
 import { BatchUpgradeModal } from './BatchUpgradeModal'
+import { useAgentClusterInstall } from '../CreateCluster/components/assisted-installer/utils'
+import {  clusterImageSetsState } from '../../../../../atoms'
+import { CIM } from 'openshift-assisted-ui-lib'
+
+const {
+    getVersionFromReleaseImage,
+} = CIM
 
 export function DistributionField(props: { cluster?: Cluster; clusterCurator?: ClusterCurator | undefined }) {
     const { t } = useTranslation()
     const [open, toggleOpen] = useState<boolean>(false)
     const toggle = () => toggleOpen(!open)
     const [ansibleJobs] = useRecoilState(ansibleJobState)
+    const [clusterImageSets] = useRecoilValue(waitForAll([clusterImageSetsState]))
+    const agentClusterInstall = useAgentClusterInstall({
+        name: props.cluster?.name!,
+        namespace: props.cluster?.namespace!,
+    })
     let latestAnsibleJob: { prehook: AnsibleJob | undefined; posthook: AnsibleJob | undefined }
     if (props.cluster?.namespace && ansibleJobs)
         latestAnsibleJob = getLatestAnsibleJob(ansibleJobs, props.cluster?.namespace)
     else latestAnsibleJob = { prehook: undefined, posthook: undefined }
 
-    if (!props.cluster?.distribution) return <>-</>
+    if (!props.cluster?.distribution) {
+        if (props.cluster?.status === ClusterStatus.draft) {//cluster is in draft we try to get version from clusterimage
+            const clusterImage = clusterImageSets.find((clusterImageSet) => clusterImageSet.metadata?.name == agentClusterInstall.spec?.imageSetRef?.name)   
+            const version =  getVersionFromReleaseImage(clusterImage?.spec?.releaseImage)                     
+            return <>{version ? `OpenShift ${version}` : '-' }</>
+        }
+        else {
+            return <>-</>
+        }
```

Comment:
How about changing this to not to be limited only to the draft status? Simply if `props.cluster.distribution` does not exist, we look at `agentClusterInstall`, try to find the `clusterImage` and get version from the release image. Then:
```
if (version) {
  return <>{version ? `OpenShift ${version}` : '-' }</>
} else {
  return <>-</>
}

Reply:
ok, is a good idea. I'll change it

Reply:
changed!